torch/jit/__init__.py

```diff
@@ -202,3 +204,23 @@ def _hide_source_ranges() -> Iterator[None]:
 
 if not torch._C._jit_init():
     raise RuntimeError("JIT initialization failed")
+
+class strict_fusion(object):
+    """
+    This class errors if not all nodes have been fused in
+    inference, or symbolically differentiated in training.
+    Example:
+    with torch.jit.strict_fusion():
+                return x + x + x
+    """
+
+    def __init__(self):
+        if not torch._jit_internal.is_scripting():
+            warnings.warn("Only works in script mode")
+        pass
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, type: Any, value: Any, tb: Any) -> bool:
+        return True
```

Comment:
why is this returning true?

from [python docs](https://docs.python.org/3/reference/datamodel.html#context-managers)
> object.__exit__(self, exc_type, exc_value, traceback)[¶](https://docs.python.org/3/reference/datamodel.html#object.__exit__)
> Exit the runtime context related to this object. The parameters describe the exception that caused the context to be exited. If the context was exited without an exception, all three arguments will be [None](https://docs.python.org/3/library/constants.html#None).
> If an exception is supplied, and the method wishes to suppress the exception (i.e., prevent it from being propagated), it should return a true value. Otherwise, the exception will be processed normally upon exit from this method.

I think this means we should return false (or None) so that in eager mode, exceptions are passed through - does that sound right?